### DISCUSSION TITLE
Trouble with ROS 2 Galactic on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by shamlian** [iRobotEducation/create3_docs#98] [Questions] [answered]
Hello, 

I have a new Create-3 robot that I am trying to hook up with a Raspberry Pi 4 and Ubuntu. 
I am following directions on https://iroboteducation.github.io/create3_docs/setup/pi4/#fn:4.  
I am up to Step 8, and the ”then” section where I get the following errors (by simply copying and executing the code):

E: Unable to locate package ros-galactic-ros-base 
E: Unable to locate package ros-galactic-rmw-fastrtps-cpp 
E: Unable to locate package ros-galactic-rmw-cyclonedds-cpp 
E: Unable to locate package ros-galactic-irobot-create-msgs

Suggestions please?
please forgive my naivety -- I am new to this. 
Many thanks.

_Originally posted by @Tokolosh4 in https://github.com/iRobotEducation/create3_docs/discussions/16#discussioncomment-3066659_

**Comment by shamlian** [comment, marked as answer]
ROS 2 Galactic is only available in apt on Ubuntu 20.04; if you are attempting these instructions in another version of Ubuntu, they will not work. Please see https://ros.org/reps/rep-2000.html#galactic-geochelone-may-2021-november-2022 for further details.

**Reply by Tokolosh4** [reply]
Thanks muchly!!

**Reply by alsora** [reply]
Note that although the debian packages are not available on ubuntu 22.04 (i.e. you can't do `sudo apt-get install ros-galactic-*`), it is still possible to build ROS 2 Galactic from sources on this platform.
This is definitely a more advanced and complex procedure, you can find some instructions here: https://docs.ros.org/en/galactic/Installation/Alternatives/Ubuntu-Development-Setup.html
